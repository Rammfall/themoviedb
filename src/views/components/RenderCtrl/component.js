import PropTypes from 'prop-types'

const RenderCtrl = ({
  isLoading,
  isEmpty,
  renderLoading,
  renderEmpty,
  renderList
}) => {
  if (isLoading) {
    return renderLoading
  }
  if (!isLoading && isEmpty) {
    return renderEmpty
  }
  return renderList
}

RenderCtrl.propTypes = {
  isLoading: PropTypes.bool,
  isEmpty: PropTypes.bool.isRequired,
  renderLoading: PropTypes.element.isRequired,
  renderEmpty: PropTypes.element.isRequired,
  renderList: PropTypes.element.isRequired
}

RenderCtrl.defaultProps = {
  isLoading: null
}

export default RenderCtrl
