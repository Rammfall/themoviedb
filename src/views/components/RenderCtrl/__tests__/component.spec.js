import React from 'react'
import { shallow } from 'enzyme'

import RenderCtrl from '../component'

describe('RenderCtrl', () => {
  const Element = <div>Test</div>
  const defaultProps = {
    isLoading: false,
    isEmpty: false,
    renderLoading: <Element />,
    renderEmpty: <Element />,
    renderList: <Element />
  }

  describe('with default props', () => {
    it('matches snapshot', () => {
      const component = shallow(<RenderCtrl {...defaultProps} />)

      expect(component).toMatchSnapshot()
    })
  })

  describe('when isLoading true', () => {
    it('matches snapshot', () => {
      const props = {
        ...defaultProps,
        isLoading: true
      }
      const component = shallow(<RenderCtrl {...props} />)

      expect(component).toMatchSnapshot()
    })
  })

  describe('when isEmpty true', () => {
    it('matches snapshot', () => {
      const props = {
        ...defaultProps,
        isEmpty: true
      }
      const component = shallow(<RenderCtrl {...props} />)

      expect(component).toMatchSnapshot()
    })
  })
})
