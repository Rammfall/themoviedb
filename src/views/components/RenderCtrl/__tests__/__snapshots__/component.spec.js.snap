// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderCtrl when isEmpty true matches snapshot 1`] = `<Element />`;

exports[`RenderCtrl when isLoading true matches snapshot 1`] = `<Element />`;

exports[`RenderCtrl with default props matches snapshot 1`] = `<Element />`;
